### PR TITLE
Add table master journal test

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang PTAL

**Comment by yuzhu** [comment]
jenkins, test this please

File: tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+public class TableMasterJournalIntegrationTest {
+  @Rule
+  public LocalAlluxioClusterResource mClusterResource =
+      new LocalAlluxioClusterResource.Builder().setStartCluster(false)
+          .build();
+
+  private static final String DB_NAME = "testdb";
+  private static final String TABLE_NAME_PREFIX = "testtable";
+
+  private void createMockUdbAndTable(int numOfTables) throws Exception {
```
Comment (gpang) [inline tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: would it be simpler to just modify the udb registry class to allow tests to set entries in the registry? Or, be able to pass in a registry to the master classes?

File: tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    PowerMockito.whenNew(UnderDatabaseRegistry.class).withNoArguments().thenReturn(registry);
+  }
+
+  @Before
+  public void before() throws Exception {
+    createMockUdbAndTable(1);
+  }
+
+  @Test
+  public void journalAttachDb() throws Exception {
```
Comment (gpang) [inline tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: Shouldn't we test additional operations like detach, new layouts, etc?

File: tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java (hunk 1 above)
Comment (yuzhu) [reply tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: the difficulties with that is all the journal operation happens when TableMaster class is started. That's why I mocked the constructor.  Passing in a registry would mean a significant change to `LocalAlluxioClusterResource` and the startMasters process.. which is not ideal either.

File: tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java (hunk 2 above)
Comment (yuzhu) [reply tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: yeah that will come once we settle on the testing infrastructure.

**Comment by chcn** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
@gpang PTAL, comments addressed.

File: tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  private static final String DB_NAME = TestDatabase.DB_NAME;
+
+  @Before
+  public void before() throws Exception {
+    TestDatabase.genTable(1);
+  }
+
+  @Test
+  public void journalAttachandDetachDb() throws Exception {
```
Comment (gpang) [inline tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: can we split up the tests so there is a test for "attach" (attach , restart, check), a test for "detach" (attach, detach, restart, check), and a test for layout changes?

Comment (yuzhu) [reply tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java]: split into attach and detach
@chcn will add the test for layout changes

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
@gpang PTAL, thanks.

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by yuzhu** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**